UI/Controller.js

```diff
@@ -0,0 +1,280 @@
+            return true;
+        }
+    }
+
+    function downLoadMore(event) {
+        if (event.target.id === "down-more") {
+            skip1 = top1;
+            let lenPosts = Posts.getLenghtPosts() - skip1;
+            top1 += 10;
+            if (lenPosts < 10) {
+                top1 -= 10;
+                top1 += lenPosts;
+            }
+
+            let morePosts = Posts.getDownMore(skip1, top1);
+            if (morePosts.length != 0) {
+                ViewPhoto.showPosts(Posts.getDownMore(skip1, top1));
+            }
+        }
+    }
+
+    function doLike(IdPost) {
+        let flag = Posts.addLikeToPost(IdPost, user);
+        
+        ViewPhoto.doLike(IdPost, flag);
+        Posts.save();
+    }
+
+    function editPosts(id) {
+        let oldPost = Posts.getPhotoPost(id);
+        const editPost = document.getElementById('dia-edit');
+        ViewPhoto.pageEditPost(oldPost, user);
+
+        if (editPost.open != true)
+            editPost.showModal();
+    }
+
+    function saveEdit (id)
+    {
+        let newPost =
+        {
+            descriprion: '',
+            photoLink: '',
+            hashtags: []
+        }
+        newPost.photoLink = document.getElementById('new-photo').value;
+        newPost.descriprion = document.getElementById('new-decr').value;
+        newPost.hashtags = document.getElementById('new-hash').value.split(',');
+
+        if (!Posts.editPhotoPost(id, newPost)) {
+            return false;
+        }
+        ViewPhoto._editPost(id, Posts.getPhotoPost(id));
+        document.getElementById('dia-edit').close();
+        Posts.save();
+        
+    }
+
+    function doDel(id) {
+          
+        if (!Posts.removePhotoPost(id)) {
+            return false;
+        }
+        skip1 -= 1;
+        top1 -= 1;
+        ViewPhoto._removePost(id);
+      
+        Posts.save();
+        return true;
+    }
+
+    function eventPost() {
+        const postsContaier = document.querySelector('main');
+        postsContaier.addEventListener('click', (event) => {
+            const target = event.target.closest('[data-action]');
+            if (event.target.id == 'like' || event.target.id == 'edit' || event.target.id == 'delete'
+            || event.target.id =='btn-edit') {
+                let post;
+                let postId;
+                if (event.target.id !='btn-edit')
+                {
+                    post = target.closest('.userphotopost');
+                    postId = post.id;
+                }
+                let prevent = false;
+                switch (target.dataset.action) {
+                    case 'like':
+                        doLike(postId);
+                        prevent = true;
+                        break;
+                    case 'edit':
+                        editPosts(postId);
+                        prevent = true;
+                        saveId = postId;
+                        break;
+                    case 'btn-edit': 
```

Comment:
What's the difference with the previous one?

Reply:
If it's handle dialog button then just handle it in dialog handler and not in the post item one )